sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -2315,6 +2315,34 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-40988: Test case for insert partition should verify value") {
+    val tableName = "partition_table"
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (id INT, name STRING) USING PARQUET PARTITIONED BY (age INT)")
```

Comment:
`SOTRED AS` rather than `USING `, otherwise it's always a datasource table

Reply:
Sure i will modify and resend the PR.